syntax = "proto3";

message HandVsRandomEquityRequest {
  string hand = 1;
}

message HandVsHandEquityRequest {
  string hero_hand = 1;
  string villain_hand = 2;
}

message HandVsRangeEquityRequest {
  string hand = 1;
  string range = 2;
}

message RangeHeatmapRequest {
}

message HandVsRangeHeatmapRequest {
  string hand = 1;
}

message SingleEquityResponse {
  float equity = 1;
}

message HeatmapResponse {
  repeated string hands = 1;
  repeated float equities = 2;
}


service EquityCalculatorService {
  rpc CalculateHandVsRandomEquity(HandVsRandomEquityRequest) returns (SingleEquityResponse);
  rpc CalculateHandVsHandEquity(HandVsHandEquityRequest) returns (SingleEquityResponse);
  rpc CalculateHandVsRangeEquity(HandVsRangeEquityRequest) returns (SingleEquityResponse);

  rpc GenerateRangeHeatmap(RangeHeatmapRequest) returns (HeatmapResponse);
  rpc GenerateHandVsRangeHeatmap(HandVsRangeHeatmapRequest) returns (HeatmapResponse);
}
